spaceship/lib/spaceship/client.rb

```diff
@@ -880,10 +880,7 @@ def send_request(method, url_or_path, params, headers, &block)
         response = @client.send(method, url_or_path, params, headers, &block)
         log_response(method, url_or_path, response, headers, &block)
 
-        resp_hash = response.to_hash
-        if resp_hash[:status] == 401
-          handle_401(response)
-        end
+        handle_error(response)
```

Comment:
Moved All error handling into `handle_error` method which can be overridden by other clients (like the connect client)